### PR TITLE
Added sum combiner (#500)

### PR DESCRIPTION
**Pull request by hletrd**
Please have a look at the following list and replace the "[ ]" with a "[x]" if
the answer to this question is yes.

- [x] For new contributors: Did you add yourself to the "Authors.rst" file?

For documentation changes:

- [ ] For documentation changes: Does your commit message include a "[skip ci]"?
      Note that it should not if you changed any examples!

For bugfixes:

- [ ] Did you add an entry to the "Changes.rst" file?
- [ ] Did you add a regression test?
- [ ] Does the commit message include a "Fixes #issue_number" (replace "issue_number").
- [ ] Does this PR add, rename, move or remove any existing functions or parameters?

For new functionality:

- [x] Did you add an entry to the "Changes.rst" file?
- [x] Did you include a meaningful docstring with Parameters, Returns and Examples?
- [x] Does the commit message include a "Fixes #issue_number" (replace "issue_number").
- [x] Did you include tests for the new functionality?
- [ ] Does this PR add, rename, move or remove any existing functions or parameters?

Please note that the last point is not a requirement. It is meant as a check if
the pull request potentially breaks backwards-compatibility.

-----------------------------------------

### CODE REVIEW

**Comment by hletrd** [comment]
Appveyor build fails on Python 3.6 because it fails to build Scipy. (Appveyor build and test on Python 2.7 succeed)

File: CHANGES.rst [hunk 1]
```diff
@@ -26,6 +26,8 @@ New Features
 
 - ``combine`` now accepts ``numpy.ndarray`` as the input ``img_list``. [#493, #503]
 
+- Added ```sum``` option in method for ```combime```. [#500]
```
Comment (MSeifert04) [inline CHANGES.rst]: Could you also include the PR number `[#500, #508]`?

File: ccdproc/combiner.py [hunk 2]
```diff
@@ -447,6 +447,73 @@ def average_combine(self, scale_func=ma.average, scale_to=None,
...
 
+    def sum_combine(self, sum_func=ma.sum, scale_to=None,
+                        uncertainty_func=ma.std):
+        """
+        Sum combine together a set of arrays.
+
+        A `~ccdproc.CCDData` object is returned with the data property
+        set to the sum of the arrays. If the data was masked or any
+        data have been rejected, those pixels will not be included in the
+        sum. A mask will be returned, and if a pixel has been
```
Comment (MSeifert04) [inline ccdproc/combiner.py]: Maybe include a sentence that if a re-scaled average sum is needed one should do `average_combine` and multiply it with the number of images combined.

File: ccdproc/combiner.py [hunk 3]
```diff
@@ -447,6 +447,73 @@ def average_combine(self, scale_func=ma.average, scale_to=None,
...
+        data = sum_func(scalings * self.data_arr, axis=0)
+
+        # set up the mask
+        masked_values = self.data_arr.mask.sum(axis=0)
+        mask = (masked_values == len(self.data_arr))
+
+        # set up the deviation
+        uncertainty = uncertainty_func(self.data_arr, axis=0)
+        # Divide uncertainty by the number of pixel (#309)
+        # Commented out because it cancels the effect of 'Multiply uncertainty by square root of the number of images'
```
Comment (MSeifert04) [inline ccdproc/combiner.py]: I think there should be another line split. Even though 80 is just a "soft limit" I see no reason to make it much longer here.

**Comment by hletrd** [comment]
* Uncertainty test for partially masked image added in 2a7e2e6
* Modified comments in 759e899

**Comment by hletrd** [comment]
AppVeyor build only fails on Python 3.6 because it fails to build astroscrappy.

**Comment by MSeifert04** [comment]
There seem to be 3 trailing whitespaces:

```
ccdproc/combiner.py:462:78: W291 trailing whitespace
        Because sum_combine returns 'pure sum' with masked pixels ignored, if
                                                                             ^
    Trailing whitespace is superfluous.


ccdproc/combiner.py:463:70: W291 trailing whitespace
        re-scaled sum is needed, average_combine have to be used with
                                                                     ^
    Trailing whitespace is superfluous.



ccdproc/combiner.py:507:77: W291 trailing whitespace
        # Commented out because it cancels the effect of 'Divide uncertainty
                                                                            ^
    Trailing whitespace is superfluous.
```

**Comment by MSeifert04** [comment]
I don't think the uncertainty is working correctly. Just a quick test shows that adding two ccds produces a different final uncertainty than sum combining the same two ccds:

```
from ccdproc import CCDData, Combiner
import numpy as np
from astropy.nddata import StdDevUncertainty

ccd1 = CCDData(np.arange(10), unit='', uncertainty=StdDevUncertainty(np.ones(10)))
ccd2 = CCDData(np.ones(10), unit='', uncertainty=StdDevUncertainty(np.ones(10)))
ccd1.add(ccd2).uncertainty
# StdDevUncertainty([ 1.41421356,  1.41421356,  1.41421356,  1.41421356,
#                     1.41421356,  1.41421356,  1.41421356,  1.41421356, 
#                     1.41421356,  1.41421356])

Combiner([ccd1, ccd2]).sum_combine().uncertainty
# StdDevUncertainty([ 0.5,  0. ,  0.5,  1. ,  1.5,  2. ,  2.5,  3. ,  3.5,   4. ])
```

In theory the final uncertainties should be identical, right?

**Comment by MSeifert04** [comment]
I'm still a bit confused about uncertainty for `sum_combine`. For average and median the uncertainty is the standard deviation or median absolute deviation but for summation (at least if it's not the rescaled average) it's a bit harder. Should it follow the normal addition error propagation or should it be the standard deviation times the number of valid pixels?

In this simple example it seems a bit inconsistent how the final uncertainty is calculated:

```
from ccdproc import CCDData, Combiner
import numpy as np
from astropy.nddata import StdDevUncertainty

ccd1 = CCDData(np.arange(10), unit='', uncertainty=StdDevUncertainty(np.ones(10)))
ccd2 = CCDData(np.ones(10), unit='', uncertainty=StdDevUncertainty(np.ones(10)))

Combiner([ccd1, ccd2]).average_combine().uncertainty
StdDevUncertainty([ 0.35355339,  0.        ,  0.35355339,  0.70710678,
                    1.06066017,  1.41421356,  1.76776695,  2.12132034,
                    2.47487373,  2.82842712])

Combiner([ccd1, ccd2]).sum_combine().uncertainty
StdDevUncertainty([ 0.5,  0. ,  0.5,  1. ,  1.5,  2. ,  2.5,  3. ,  3.5,
                    4. ])
```

**Comment by hletrd** [comment]
I was wrong.

**Comment by crawfordsm** [comment]
Thanks @hletrd !

**Pull request merged**